### PR TITLE
BaSP-TG-85: Update tasks forms

### PR DESCRIPTION
**Pull request by LuliBianchini**
### Tasks done in the PR:
- Implementation of React Hook Form in Edit and Add Tasks Forms
- Implementation of Joi validations in both forms 

### Expected result:
- Correct visualization and working of forms with their corresponding inputs

### CODE REVIEW

File: src/Components/Tasks/Edit/Edit.jsx [hunk 1]
```diff
@@ -1,98 +1,132 @@
...
+  startDate: Joi.date().required().messages({ 'any.required': 'Start date is a required field' }),
+  workedHours: Joi.number().integer().min(0).required().messages({
+    'number.integer': 'Invalid number, it must be an integer',
+    'number.min': 'Invalid number, it must be positive',
+    'any.required': 'Worked hours is a required field'
+  }),
+  description: Joi.string().min(6).max(150).required().messages({
+    'string.min': 'Invalid description, it must contain more than 6 letters',
+    'string.max': 'Invalid description, it must not contain more than 150 letters',
+    'any.required': 'Description is a required field'
```
Comment (NicolasFontana) [inline src/Components/Tasks/Edit/Edit.jsx:33]: For the correct message to be displayed you should change 'any.required' to 'string.empty'

File: src/Components/Tasks/Edit/Edit.jsx [hunk 2]
```diff
@@ -1,98 +1,132 @@
...
+  }),
+  status: Joi.string()
+    .min(2)
+    .valid('To do', 'In progress', 'Review', 'Blocked', 'Done', 'Cancelled')
+    .required()
+    .messages({
+      'string.min': 'Invalid status, it must contain more than 2 letters',
+      'any.valid':
+        'Invalid status, it must be one of the following: To do, In progress, Review, Blocked, Done, Cancelled',
+      'any.required': 'Status is a required field'
```
Comment (NicolasFontana) [inline src/Components/Tasks/Edit/Edit.jsx:43]: For the correct message to be displayed you should change 'any.required' to 'string.empty'

File: src/Components/Tasks/Form/Form.jsx [hunk 3]
```diff
@@ -1,98 +1,133 @@
...
+    .min(2)
+    .valid('To do', 'In progress', 'Review', 'Blocked', 'Done', 'Cancelled')
+    .required()
+    .messages({
+      'string.min': 'Invalid status, it must contain more than 2 letters',
+      'any.valid':
+        'Invalid status, it must be one of the following: To do, In progress, Review, Blocked, Done, Cancelled',
+      'any.required': 'Status is a required field'
+    })
+});
```
Comment (NicolasFontana) [inline src/Components/Tasks/Form/Form.jsx:45]: You can copy and replace the task Schema with the fixes marked for the add method

**Review by NicolasFontana — CHANGES_REQUESTED** [review]

File: src/Components/Tasks/Edit/Edit.jsx [hunk 4]
```diff
@@ -1,98 +1,131 @@
...
+  const {
+    handleSubmit,
+    register,
+    formState: { errors }
+  } = useForm({
+    mode: 'onChange',
+    resolver: joiResolver(taskSchema),
+    defaultValues: {
+      taskName: task.taskName,
+      startDate: task.startDate,
```
Comment (NicolasFontana) [inline src/Components/Tasks/Edit/Edit.jsx:78]: When I click on edit, it does not load the date of that row

File: src/Components/Tasks/Edit/Edit.jsx [hunk 5]
```diff
@@ -1,98 +1,131 @@
...
+    .max(50)
+    .required()
+    .messages({
+      'string.min': 'Invalid task name, it must contain more than 3 letters',
+      'string.max': 'Invalid task name, it must not contain more than 50 letters',
+      'string.empty': 'Task name is a required field',
+      'string.pattern.base':
+        'Must contain only letters and words can only be separated by a single white space'
+    }),
+  startDate: Joi.date().required().messages({ 'any.required': 'Start date is a required field' }),
```
Comment (NicolasFontana) [inline src/Components/Tasks/Edit/Edit.jsx:23]: You have to change `'any.required'` for this `'string.empty'`

File: src/Components/Tasks/Form/Form.jsx [hunk 6]
```diff
@@ -1,98 +1,132 @@
...
+    .max(50)
+    .required()
+    .messages({
+      'string.min': 'Invalid task name, it must contain more than 3 letters',
+      'string.max': 'Invalid task name, it must not contain more than 50 letters',
+      'string.empty': 'Task name is a required field',
+      'string.pattern.base':
+        'Must contain only letters and words can only be separated by a single white space'
+    }),
+  startDate: Joi.date().required().messages({ 'any.required': 'Start date is a required field' }),
```
Comment (NicolasFontana) [inline src/Components/Tasks/Form/Form.jsx:23]: You have to change 'any.required' for this 'string.empty'

File: src/Components/Tasks/Form/Form.jsx [hunk 7]
```diff
@@ -1,98 +1,132 @@
...
+  }),
+  status: Joi.string()
+    .min(2)
+    .valid('To do', 'In progress', 'Review', 'Blocked', 'Done', 'Cancelled')
+    .required()
+    .messages({
+      'string.min': 'Invalid status, it must contain more than 2 letters',
+      'any.valid':
+        'Invalid status, it must be one of the following: To do, In progress, Review, Blocked, Done, Cancelled',
+      'string.empty': 'Status is a required field'
```
Comment (NicolasFontana) [inline src/Components/Tasks/Form/Form.jsx:42]: 'string.empty' message is not working

**Review by NicolasFontana — CHANGES_REQUESTED** [review]

**Review by NicolasFontana — APPROVED** [review]
Everything works well!

**Comment by LeonelListro** [comment]
# Test results

Tested locally on: 

test result: **failed** ⚠️

## bugs found:

1- The task name field should allow be completed with numbers too.
task name

2- The validation error message of the "Worked hours" field is the same when you let the empty field and when you fill it whit letters. 
correct message

**Review by LeonelListro — CHANGES_REQUESTED** [review]
Please refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/43#issuecomment-1165933104

**Review by LeonelListro — APPROVED** [review]
Well done Lucia and team!

**Pull request merged**